pallets/asset/src/lib.rs

```diff
@@ -365,17 +365,24 @@ decl_module! {
             use polymesh_primitives::migrate::{migrate_double_map, Migrate};
             use polymesh_primitives::document::DocumentOld;
             use sp_std::collections::btree_map::BTreeMap;
-            let mut id_map = BTreeMap::<_, u32>::new();
-            migrate_double_map::<_, _, Blake2_128Concat, _, _, _, _, _>(
+
+            let spec_version = frame_system::LastRuntimeUpgrade::get().map_or(0, |upgrade| upgrade.spec_version.0);
+
+            if spec_version == 2000
```

Comment:
This won't work because Alcyone has a higher spec version.

Reply:
Also, the hard coded constant is suboptimal. As a general note, the copy pasted `let spec_version` in each file should really be deduped into one place.

Reply:
New logic based on local `StorageVersion`.